src/web100srv.c

```diff
@@ -1443,11 +1425,861 @@ int run_test(tcp_stat_agent* agent, int ctlsockfd, TestOptions* testopt,
              vars.Timeouts, vars.Sndbuf, vars.MaxRwinRcvd, vars.CurrentCwnd,
              mismatch, bad_cable, totalcnt, refresh);
   }
-  shutdown(ctlsockfd, SHUT_WR);
+  if (ctl->ssl != NULL) {
+    SSL_shutdown(ctl->ssl);
+  } else {
+    shutdown(ctl->socket, SHUT_WR);
+  }
```

Comment:
Why not `close_connection` here? Also, should `close_connection` also be calling `shutdown` in the case of non-TLS connections?